### PR TITLE
api-reference: publish doc only on api changes

### PR DESCRIPTION
**Pull request by lukas-bednar**
generate readme after api-changes were detected.

Fixes: kubevirt-incubator/api-reference#6

### CODE REVIEW

**Comment by rmohr** [comment]
@lukas-bednar my suspicion would be that it gets updated because of the version change in the docs and not because of the readme. The readme should always consume the already existing README.md if no change exists in the repo. Is that taken into account?

**Comment by lukas-bednar** [comment]
@rmohr ah yes, you are right, I am totally off from the real problem ...
Thanks!

File: hack/gen-swagger-doc/deploy.sh [hunk 1]
```diff
@@ -38,7 +38,8 @@ find * -type d -regex "^v[0-9.]*" \
 git config --global user.email "travis@travis-ci.org"
 git config --global user.name "Travis CI"
 
-if git status --porcelain | grep --quiet "^ [AM]"; then
+# NOTE: exclude index.html from match, becasue it is static except commit hash.
```
Comment (lukas-bednar) [inline hack/gen-swagger-doc/deploy.sh]: I couldn't come with something better. If you have any better idea, I will be glad to hear!

**Comment by fabiand** [comment]
Tada

**Pull request merged**